java/src/jmri/configurexml/ConfigXmlManager.java

```diff
@@ -121,23 +121,22 @@ public Object findInstance(Class<?> c, int index) {
 
     @Override
     public ArrayList<Object> getInstanceList(Class<?> c) {
+        ArrayList<Object> result = new ArrayList<>();
+
         ArrayList<Object> temp = new ArrayList<>(plist);
-        ArrayList<Object> returnlist = new ArrayList<>();
         temp.addAll(clist.keySet());
         temp.addAll(tlist);
         temp.addAll(ulist);
         temp.addAll(uplist);
-        for (int i = 0; i < temp.size(); i++) {
-            if (c.isInstance(temp.get(i))) {
-                returnlist.add(temp.get(i));
-                //if (index-- == 0) return temp.get(i);
+        for (Object o : temp) {
+            if (c.isInstance(o)) {
+                result.add(o);
             }
         }
-        if (returnlist.isEmpty()) {
-            return null;
+        if (result.isEmpty()) {
+            result = null;
         }
-        return returnlist;
-
+        return result;
```

Comment:
When returning a Collection (or Set, List, HashMap, etc), it is almost *always* better to return an empty list than null if the list is empty (there is even a FindBugs target to trap that).

Returning an empty list avoids the need to set the list null if empty in the returning method, but also means that the results of the method can be marked ```@Nonnull``` and consuming code does not need to check that the list is null.

The difference for the consuming code can be:
```
for (Object o : getInstanceList(Foo.class)) {
    ...
}
```
vs:
```
List<Object> l = getInstanceList(Foo.class);
if (l != null) {
    for (Object o : l) {
        ...
    }
}
```

Reply:
That code has
`        if (result.isEmpty()) {
            result = null;
        }
`
Is there something else you want there?

Reply:
Oh, never mind; I read that backwards… I'll remove the isEmpty conditional.